homeassistant/components/tilt_pi/config_flow.py

```diff
@@ -0,0 +1,71 @@
+"""Config flow for Tilt Pi integration."""
+
+import logging
+from typing import Any
+
+import aiohttp
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PORT
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .api import TiltPiClient, TiltPiError
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME): str,
+        vol.Required(CONF_HOST): str,
+        vol.Optional(CONF_PORT, default=1880): int,
+    }
+)
+
+
+class TiltPiConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Tilt Pi."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle a configuration flow initialized by the user."""
+
+        if user_input is not None:
+            await self.async_set_unique_id(f"tiltpi_{user_input[CONF_HOST]}")
+            self._abort_if_unique_id_configured()
```

Comment:
the host is a bad unique id, it should be truly unique, otherwise use `async_abort_entries_match`